<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
    <meta http-equiv="x-ua-compatible" content="ie=edge" />
    <meta name="theme-color" content="#e87860">
    <meta name="description" content="Our research aims to analyse biomedical data. We use theory coming from machine learning, data science, mathematics and graph theory." />
    <meta name="keywords" content="InfOmics data omics biomedical univr research software" />
    <title>Softwares</title>
    <link rel="icon" href="../img/homepage/logo.ico">
    <!-- Font Awesome -->
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.11.2/css/all.css" />
    <!-- Google Fonts Roboto -->
    <link
      rel="stylesheet"
      href="https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500;700&display=swap"
    />
    <!-- MDB -->
    <link rel="stylesheet" href="../css/mdb.min.css" />
    <link rel="stylesheet" href="../css/custom.css" />
    <script src='https://api.mapbox.com/mapbox-gl-js/v2.1.0/mapbox-gl.js'></script>
  <link href='https://api.mapbox.com/mapbox-gl-js/v2.1.0/mapbox-gl.css' rel='stylesheet' />
  </head>
  <body>
    <div class="head-section head-section-software">
    <!-- Navbar -->
      <nav class="navbar navbar-expand-lg">
      <!-- Container wrapper -->
      <div class="container-fluid ">
        <!-- Navbar brand -->
        <a class="navbar-brand" href="../index.html">
          <img
            src="../img/homepage/Infoverona.png"
            height="100"
            alt="Lab Logo"
          />
        </a>

        <!-- Toggle button -->
        <button
          class="navbar-toggler"
          type="button"
          data-mdb-toggle="collapse"
          data-mdb-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <i class="fas fa-bars"></i>
        </button>

        <!-- Collapsible wrapper -->
        <div class="collapse navbar-collapse" id="navbarSupportedContent">
          <!-- Left links -->
          <ul class="navbar-nav me-auto mb-2 mb-lg-0">
            <li class="nav-item">
              <a class="nav-link active" aria-current="page" href="../index.html">HOMEPAGE</a>
            </li>
            <li class="nav-item">
              <a class="nav-link active" href="./people.html">PEOPLE</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="./publications.html">PUBLICATIONS</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="#">SOFTWARE</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="./lab_activities.html">LAB ACTIVITIES</a>
            </li>
            <li class="nav-item github">
              <a class="nav-link" href="https://github.com/InfOmics" target="_blank">
                <span><i class="fab fa-github"></i></span>
              </a>
            </li>
          </ul>
          
        </div>

        <!-- Collapsible wrapper -->
      </div>
      <!-- Container wrapper -->
      </nav>
      <!-- Navbar -->
      <div class="container-fluid presentation">
        <h1>Selected Software</h1>
        <div class="presentation-text">
          <p>
            Below are listed the main software developed and published to 
            bioinformatics community by InfOmics lab members. For a comprehensive 
            and complete list of software developed by our lab click
            <a href="https://github.com/InfOmics"
              target="_blank" rel="noreferrer">here</a>
          </p> 
        </div>
      </div>
    </div>

    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>APBIO</h2>
            <p>APBIO is a tool developed to predict compound-target interactions, with a specific focus
               on air pollutants and their bioactivity. It computes bioactivity signatures for compounds
               starting from the SMILES representation (e.g., C1=CC=CC=C1) and FASTA sequence features
               for targets starting from the UniProtKB identifier (e.g., Q9UHW9). This allows unknown ligands
               to be prioritized for a given target (or vice versa), and subsequently tested through molecular
               docking simulations or validated via in vitro experiments for further studies on gene expression,
               chemical toxicity, and hazard evaluation.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/evaviesi/APBIO/tree/main" 
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/evaviesi/APBIO/archive/refs/heads/main.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/APBIO.jpeg"
            alt="APBIO.jpeg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>EasyCircR</h2>
            <p>EasyCircR is a novel R package that combines circRNA detection and reconstruction
               with post-transcriptional gene expression analysis (exon-intron split analysis)
               and miRNA response element prediction. The package enables estimation and visualization
               of circRNA-miRNA-mRNA interactions through an intuitive Shiny application, leveraging
               the post-transcriptional regulatory nature of circRNA-miRNA relationship and excluding unrealistic
               regulatory interactions at the biological level.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="http://github.com/InfOmics/EasyCircR" 
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/EasyCircR/archive/refs/heads/main.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/EasyCIrcR.png"
            alt="EasyCircR.png"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>CRISPRme</h2>
            <p>CRISPRme is a tool for comprehensive off-target assessment available 
              as a web application online, offline, and command line. It integrates 
              human genetic variant datasets with orthogonal genomic annotations to 
              predict and prioritize CRISPR-Cas off-target sites at scale. The method 
              considers both single-nucleotide variants (SNVs) and indels, accounts for 
              bona fide haplotypes, accepts spacer:protospacer mismatches and bulges, 
              and is suitable for population and personal genome analyses. CRISPRme 
              takes care of all steps in the process including data download, executing 
              the complete search, and presents an exhaustive report with tables and 
              figures within interactive web-based GUI.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/pinellolab/CRISPRme" 
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/pinellolab/CRISPRme/archive/refs/heads/main.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/crisprme.jpg"
            alt="crisprme.jpg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>GRAFIMO</h2>
            <p>GRAFIMO (GRAph-based Finding of Individual Motif Occurrences) is 
              a command-line tool that extends the traditional Position Weight Matrix 
              (PWM) scanning procedure to pangenome variation graphs (VGs). GRAFIMO 
              can search the occurrences of a given PWM in many genomes in a single 
              run, accounting for the effects that SNPs, indels and potentially 
              any structural variation (handled by VG) have on found potential 
              motif occurrences. As result, GRAFIMO produces a report containing 
              the statistically significant motif candidates found, reporting their 
              frequency within the haplotypes embedded in the scanned VG and if they 
              contain genomic variants or belong to the reference genome sequence.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/pinellolab/GRAFIMO"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/pinellolab/GRAFIMO/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/grafimo.jpg"
            alt="grafimo.jpg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>APDB</h2>
            <p>APDB is a user-friendly platform for the inspection of collected 
              air pollutants and the search for similar molecules in the investigated 
              chemical spaces. The web interface allows (i) to browse data by category, 
              (ii) to query the database by molecules and targets, and (iii) to 
              visualize and download molecular descriptors and structures as well as 
              computed similarities.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/APDB"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="http://apdb.di.univr.it/apdb/home" 
                target="_blank">View Online</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/apdb.png"
            alt="apdb.png"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>Stardust</h2>
            <p>Stardust is a clustering algorithm for Spatial Transcriptomics data.
              Its aim is, given as input an expression matrix, 
              the positions of spots and a space weight configuration, to derive a 
              vector of cluster identities for each spot in the input data. The similarity between spots is 
              composed by integrating transcripts and physical distance of spots in a user-driven fashion.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/stardust"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/stardust/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/stardust.png"
            alt="stardust.jpg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>Esearch3D</h2>
            <p>Signals pertaining to transcriptional activation are transferred 
              from enhancers found in intergenic loci to genes in the form of 
              transcription factors, cofactors, and various transcriptional machinery 
              such as RNA Pol II. How and where this information is transmitted to 
              and from is central for decoding the regulatory landscape of any gene 
              and identifying enhancers. Esearch3D is an unsupervised algorithm designed 
              to predict enhancer regions from 3C data. It works by reverse engineering 
              the flow of regulatory information from enhancers to genes. Using this 
              approach we were able to identify intergenic regulatory enhancers using 
              solely gene expression and 3D genomic data.
              Esearch3D models chromosome conformation capture (3C) data as chromatin 
              interaction network (CIN) and then exploits graph-theory algorithms to 
              integrate RNA-seq data to calculate an imputed activity score (IAS) for 
              intergenic regions. We also provide a visualisation tool to allow an easy 
              interpretation of the results.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/Esearch3D" 
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/Esearch3D/archive/refs/heads/main.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/esearch3d.png"
            alt="esearch3d.png"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>CRISPRitz</h2>
            <p>CRISPRitz is software suite to perform in-silico CRISPR analysis 
              and assessment. CRISPRitz suite contains 5 different tools 
              dedicated to perform predictive analysis and result assessement 
              on CRISPR/Cas experiments.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/CRISPRitz"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/pinellolab/CRISPRitz/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/crispritz.png"
            alt="crispritz.jpg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>LErNet</h2>
            <p>LErNet is a method to in silico define and predict the roles of 
              IncRNAs. The core of the approach is a network expansion algorithm 
              which enriches the genomic context of IncRNAs. The context is 
              built by integrating the genes encoding proteins that are found 
              next to the non-coding elements both at genomic and system level. 
              The pipeline is particularly useful in situations where the 
              functions of discovered IncRNAs are not yet known.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/LErNet"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/LErNet/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/lernet.jpg"
            alt="lernet.jpg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>PanDelos</h2>
            <p>PanDelos is a stand alone tool for the discovery of pan-genome 
              contents among phylogenetic distant genomes. The methodology is 
              based on information theory and network analysis. It is 
              parameter-free because thresholds are automatically deduced from 
              the context.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/PanDelos"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/PanDelos/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/pandelos.png"
            alt="pandelos.webp"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>PanDelos-frags</h2>
            <p>There is a plethora of incomplete genomes that are available on public resources,meaning that 
              the process for reconstructing their genomic sequence is not complete, and only fragments of 
              their sequence are currently available. However, the information contained in these fragments 
              may play an essential role in the analyses. We present PanDelos-frags, a computational tool 
              which exploits and extends previous results in analysing complete genomes of PanDelos. 
              It provides a new methodology for inferring missing genetic information and thus for managing 
              incomplete genomes. PanDelos-frags outperforms state-of-the-art approaches in reconstructing gene 
              families in synthetic benchmarks and in a real use case of metagenomics</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/PanDelos-frags" 
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/PanDelos-frags/archive/refs/heads/main.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/pandelos-frags.png"
            alt="pandelos-frags.png"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>GraphGrep-SX</h2>
            <p>GraphGrepSX is a querying system for databases of graphs. It is based 
              on its predecessor GraphGrep. The system implements efficient graph searching 
              algorithms together with advanced filtering techniques that allow an i
              nitial approximate search. It allows users to select candidate subgraphs 
              rather than entire graphs. The method searches for subgraph isomorphism 
              (monomorphism) between a query graph (pattern) and graphs inside a database 
              of graphs (targets).</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/GraphGrepSX"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/GraphGrepSX/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>cuRnet</h2>
            <p>cuRnet is a package that provides a wrap of parallel graph algorithms 
              developed in CUDA for the R environment. It makes available GPU 
              solutions to R end-users in a transparent way, by including basic 
              data structures for representing graphs, and parallel implementation 
              of of BFS (Breadth-First Search), SCC (Strongly Connected Components) 
              and SSSP (Single-Source Shortest Paths) customized for biological 
              network analysis on GPUs (Busato and Bombieri, 2016).</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/cuRnet"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/cuRnet/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/curnet.png"
            alt="curnet.webp"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>APPAGATO</h2>
            <p>APPAGATO is a stochastic and parallel algorithm to find 
              approximate occurrences of a query network in biological networks. 
              APPAGATO allows nodes and edges mismatches. To speed-up the querying 
              process, APPAGATO has been also implemented in parallel to run on 
              graphics processing units (GPUs).</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/APPAGATO"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/cuRnet/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/appagato.png"
            alt="appagato.png"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>GRAPES</h2>
            <p>GRAPES is a querying system for parallel searching in databases of graphs, 
              and single target graph, using symmetric multiprocessing (SMP) architectures. 
              It implements a parallel version of well established graph searching algorithms 
              providing efficient solutions for graphs indexing and matching. The GRAPES method 
              consists of three main phases: indexing, filtering, and matching. In the 
              indexing phase, features are extracted from target graphs and a database index 
              is built off-line. A feature is a labeled path presents in a graph. GRAPES extracts 
              all paths up to a fixed length (lp) and stores them in a compact trie structure. 
              Moreover, starting nodes of such paths are also stored. Once the index is built, 
              the system is able to find subgraph isomorphims between a query and all the target 
              graphs. The filtering phase allows to a priori discard such database graphs which 
              do not contain the query's features. It is also able to preliminarily recognize 
              unmatching nodes, and entire regions, of database graphs. This useful behavior allows 
              GRAPES to dial both with large database or single target graphs. Finally, an exact 
              subgraph matching algorithm is run in parallel. See the related scientific paper for 
              more details.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/GRAPES"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/GRAPES/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/grapes.png"
            alt="grapes.png"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>RI</h2>
            <p>RI is a general purpose algorithm for one-to-one exact subgraph 
              isomorphism problem maintaining topological constraints. It is both 
              a C++ library and a standalone tool, providing developing API and 
              a command line interface, with no dependencies out of standard GNU 
              C++ library. RI works on Unix and Mac OS X systems with G++ installed, 
              and it can be compiled under Windows using Gygwin. Working graphs may 
              be directed, undirected, multigraphs with optional attributes both on 
              nodes and edges. Customizable features allow user-defined behaviors for 
              attribute comparisons and the algorithm's flow. RI aims to provide a 
              better search strategy for the common used backtracking approach to 
              the subgraph isomorphism problem. It can be integrated with additional 
              preprocessing steps or it can be used for the verification of candidate 
              structures coming from data mining, data indexing or other filtering techniques. 
              RI is able to find graphs isomorphisms, subgraph isomorphisms and induced 
              subgraph isomorphisms. It is distributed in several versions divide chiefly in 
              two groups respectively for static or dynamically changing attributes. All 
              proposed versions are developed taking into account trade-offs between time 
              and memory requirements. Optional behaviors such as stop at first encountered 
              match, processing of result matches, type of isomorphism and additional features 
              may be enabled thanks to high modularity and library's API.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/RI"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/RI/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
          </div>
        </div>  
      </div>
    </div>
    <button onclick="topFunction()" id="myBtn" title="Go to top">Up</button>
    <!-- MDB -->
    <script type="text/javascript" src="../js/mdb.min.js"></script>
    <script type="text/javascript" src="../js/custom.js"></script>
  </body>
</html>
