<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
    <meta http-equiv="x-ua-compatible" content="ie=edge" />
    <meta name="theme-color" content="#e87860">
    <title>Softwares</title>
    <link rel="icon" href="../img/homepage/logo.ico">
    <!-- Font Awesome -->
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.11.2/css/all.css" />
    <!-- Google Fonts Roboto -->
    <link
      rel="stylesheet"
      href="https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500;700&display=swap"
    />
    <!-- MDB -->
    <link rel="stylesheet" href="../css/mdb.min.css" />
    <link rel="stylesheet" href="../css/custom.css" />
    <script src='https://api.mapbox.com/mapbox-gl-js/v2.1.0/mapbox-gl.js'></script>
  <link href='https://api.mapbox.com/mapbox-gl-js/v2.1.0/mapbox-gl.css' rel='stylesheet' />
  </head>
  <body>
    <div class="head-section head-section-software">
    <!-- Navbar -->
      <nav class="navbar navbar-expand-lg">
      <!-- Container wrapper -->
      <div class="container-fluid ">
        <!-- Navbar brand -->
        <a class="navbar-brand" href="../index.html">
          <img
            src="../img/homepage/Infoverona.png"
            height="100"
            alt="Lab Logo"
          />
        </a>

        <!-- Toggle button -->
        <button
          class="navbar-toggler"
          type="button"
          data-mdb-toggle="collapse"
          data-mdb-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <i class="fas fa-bars"></i>
        </button>

        <!-- Collapsible wrapper -->
        <div class="collapse navbar-collapse" id="navbarSupportedContent">
          <!-- Left links -->
          <ul class="navbar-nav me-auto mb-2 mb-lg-0">
            <li class="nav-item">
              <a class="nav-link active" aria-current="page" href="../index.html">HOMEPAGE</a>
            </li>
            <li class="nav-item">
              <a class="nav-link active" href="./people.html">PEOPLE</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="./publications.html">PUBLICATIONS</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="#">SOFTWARES</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="./lab_activities.html">LAB ACTIVITIES</a>
            </li>
            <li class="nav-item github">
              <a class="nav-link" href="https://github.com/InfOmics" target="_blank">
                <span><i class="fab fa-github"></i></span>
              </a>
            </li>
          </ul>
          
        </div>

        <!-- Collapsible wrapper -->
      </div>
      <!-- Container wrapper -->
      </nav>
      <!-- Navbar -->
      <div class="container-fluid presentation">
        <h1>Selected Softwares</h1>
        <div class="presentation-text">
          <p>
            Below are listed the main softwares developed and published to 
            bioinformatics community by InfOmics lab members. For a comprehensive 
            and complete list of softwares developed by our lab click
            <a href="https://github.com/InfOmics"
              target="_blank">here</a>
          </p> 
        </div>
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>CRISPRme</h2>
            <p>CRISPRme is a web based tool dedicated to perform predictive analysis 
              and result assessement on CRISPR/Cas-9 experiments with a user-friendly 
              GUI and the precise scope of searching individual variant in VCF dateset.
              With this aim in mind we created a simple package that takes care 
              of any step, from downloading the necessary data, to execute a complete 
              search and present to the user an exhaustive result report with images 
              and tabulated targets to navigate with the included web-based GUI.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/pinellolab/CRISPRme" 
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/pinellolab/CRISPRme/archive/refs/heads/main.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/crisprme.jpg"
            alt="crisprme.jpg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>GRAFIMO</h2>
            <p>GRAFIMO (GRAph-based Finding of Individual Motif Occurrences) is 
              a command-line tool that extends the traditional Position Weight Matrix 
              (PWM) scanning procedure to pangenome variation graphs (VGs). GRAFIMO 
              can search the occurrences of a given PWM in many genomes in a single 
              run, accounting for the effects that SNPs, indels and potentially 
              any structural variation (handled by VG) have on found potential 
              motif occurrences. As result, GRAFIMO produces a report containing 
              the statistically significant motif candidates found, reporting their 
              frequency within the haplotypes embedded in the scanned VG and if they 
              contain genomic variants or belong to the reference genome sequence.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/pinellolab/GRAFIMO"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/pinellolab/GRAFIMO/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/grafimo.jpg"
            alt="grafimo.jpg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>Stardust</h2>
            <p>Stardust in a clustering algorithm for Spatial Transcriptomics data.
              Its aim is, given as input an expression matrix, 
              the positions of spots and a space weight configuration, to derive a 
              vector of cluster identities for each spot in the input data. The similarity between spots is 
              composed by integrating transcripts and physical distance of spots in a user-driven fashion.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/stardust"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/stardust/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/stardust.png"
            alt="stardust.jpg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>CRISPRitz</h2>
            <p>CRISPRitz is software suite to perform in-silico CRISPR analysis 
              and assessment. CRISPRitz suite contains 5 different tools 
              dedicated to perform predictive analysis and result assessement 
              on CRISPR/Cas experiments.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/CRISPRitz"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/pinellolab/CRISPRitz/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/crispritz.png"
            alt="crispritz.jpg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>LErNet</h2>
            <p>LErNet is a method to in silico define and predict the roles of 
              IncRNAs. The core of the approach is a network expansion algorithm 
              which enriches the genomic context of IncRNAs. The context is 
              built by integrating the genes encoding proteins that are found 
              next to the non-coding elements both at genomic and system level. 
              The pipeline is particularly useful in situations where the 
              functions of discovered IncRNAs are not yet known.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/LErNet"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/LErNet/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/lernet.jpg"
            alt="lernet.jpg"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>PanDelos</h2>
            <p>PanDelos is a stand alone tool for the discovery of pan-genome 
              contents among phylogenetic distant genomes. The methodology is 
              based on information theory and network analysis. It is 
              parameter-free because thresholds are automatically deduced from 
              the context.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/PanDelos"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/PanDelos/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/pandelos.webp"
            alt="pandelos.webp"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>cuRnet</h2>
            <p>cuRnet is a package that provides a wrap of parallel graph algorithms 
              developed in CUDA for the R environment. It makes available GPU 
              solutions to R end-users in a transparent way, by including basic 
              data structures for representing graphs, and parallel implementation 
              of of BFS (Breadth-First Search), SCC (Strongly Connected Components) 
              and SSSP (Single-Source Shortest Paths) customized for biological 
              network analysis on GPUs (Busato and Bombieri, 2016).</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/cuRnet"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/cuRnet/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/curnet.webp"
            alt="curnet.webp"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>APPAGATO</h2>
            <p>APPAGATO is a stochastic and parallel algorithm to find 
              approximate occurrences of a query network in biological networks. 
              APPAGATO allows nodes and edges mismatches. To speed-up the querying 
              process, APPAGATO has been also implemented in parallel to run on 
              graphics processing units (GPUs).</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/APPAGATO"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/cuRnet/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/appagato.png"
            alt="appagato.png"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>GRAPES</h2>
            <p>GRAPES is a querying system for parallel searching in databases of graphs, 
              and single target graph, using symmetric multiprocessing (SMP) architectures. 
              It implements a parallel version of well established graph searching algorithms 
              providing efficient solutions for graphs indexing and matching. The GRAPES method 
              consists of three main phases: indexing, filtering, and matching. In the 
              indexing phase, features are extracted from target graphs and a database index 
              is built off-line. A feature is a labeled path presents in a graph. GRAPES extracts 
              all paths up to a fixed length (lp) and stores them in a compact trie structure. 
              Moreover, starting nodes of such paths are also stored. Once the index is built, 
              the system is able to find subgraph isomorphims between a query and all the target 
              graphs. The filtering phase allows to a priori discard such database graphs which 
              do not contain the query's features. It is also able to preliminarily recognize 
              unmatching nodes, and entire regions, of database graphs. This useful behavior allows 
              GRAPES to dial both with large database or single target graphs. Finally, an exact 
              subgraph matching algorithm is run in parallel. See the related scientific paper for 
              more details.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/GRAPES"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/GRAPES/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
            <img src="../img/softwares/grapes.png"
            alt="grapes.png"/>
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>RI</h2>
            <p>RI is a general purpose algorithm for one-to-one exact subgraph 
              isomorphism problem maintaining topological constraints. It is both 
              a C++ library and a standalone tool, providing developing API and 
              a command line interface, with no dependencies out of standard GNU 
              C++ library. RI works on Unix and Mac OS X systems with G++ installed, 
              and it can be compiled under Windows using Gygwin. Working graphs may 
              be directed, undirected, multigraphs with optional attributes both on 
              nodes and edges. Customizable features allow user-defined behaviors for 
              attribute comparisons and the algorithm's flow. RI aims to provide a 
              better search strategy for the common used backtracking approach to 
              the subgraph isomorphism problem. It can be integrated with additional 
              preprocessing steps or it can be used for the verification of candidate 
              structures coming from data mining, data indexing or other filtering techniques. 
              RI is able to find graphs isomorphisms, subgraph isomorphisms and induced 
              subgraph isomorphisms. It is distributed in several versions divide chiefly in 
              two groups respectively for static or dynamically changing attributes. All 
              proposed versions are developed taking into account trade-offs between time 
              and memory requirements. Optional behaviors such as stop at first encountered 
              match, processing of result matches, type of isomorphism and additional features 
              may be enabled thanks to high modularity and library's API.</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/RI"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/RI/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
          </div>
        </div>  
      </div>
    </div>
    <div class="software-row">
      <div class="container-md">
        <div class="row">
          <div class="col-12 col-lg-4">
            <h2>GraphGrep-SX</h2>
            <p>GraphGrepSX is a querying system for databases of graphs. It is based 
              on its predecessor GraphGrep. The system implements efficient graph searching 
              algorithms together with advanced filtering techniques that allow an i
              nitial approximate search. It allows users to select candidate subgraphs 
              rather than entire graphs. The method searches for subgraph isomorphism 
              (monomorphism) between a query graph (pattern) and graphs inside a database 
              of graphs (targets).</p>
              <div class="row">
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/GraphGrepSX"
                target="_blank">Github Page</a>
                  <a class="button-size btn btn-primary" href="https://github.com/InfOmics/GraphGrepSX/archive/refs/heads/master.zip" 
                target="_blank">Download ZIP</a>
              </div>
          </div>
          <div class="col-12 col-lg-6">
          </div>
        </div>  
      </div>
    </div>
    
    <button onclick="topFunction()" id="myBtn" title="Go to top">Up</button>
    <!-- MDB -->
    <script type="text/javascript" src="../js/mdb.min.js"></script>
    <script type="text/javascript" src="../js/custom.js"></script>
  </body>
</html>